Module6/routing

yurakhomitsky opened the pull request:
- Add routing
- Refactor code

yurakhomitsky committed:
Add Routing & Refactor Code

igor-budzin commented:
There are a few issues. Please fix them so I can proceed reviewing

Issue 1
**Step to reproduce**: 
1) Select action genre
2) Select sort by title
3) Reload page

**Expected behaviour**:
Nothing changes after reload. User should see the same list of movies, URL should be the same

**Actual behaviour**:
URL is changed (seems to me to some default URL that you defined somewhere in the code)

Issue 2
By the Evaluation criteria URL "/" should also work (so in this case default values for API call should be defined somewhere in the code but not in URL)

yurakhomitsky commented:
> There are a few issues. Please fix them so I can proceed reviewing
> 
> Issue 1 **Step to reproduce**:
> 
> 1. Select action genre
> 2. Select sort by title
> 3. Reload page
> 
> **Expected behaviour**: Nothing changes after reload. User should see the same list of movies, URL should be the same
> 
> **Actual behaviour**: URL is changed (seems to me to some default URL that you defined somewhere in the code)
> 
> Issue 2 By the Evaluation criteria URL "/" should also work (so in this case default values for API call should be defined somewhere in the code but not in URL)

Thank you for the catch. Fixed and added E2E to cover those cases
